rust/agama-lib/src/questions/store.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) [2025] SUSE LLC
+//
+// All Rights Reserved.
+//
+// This program is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 2 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE LLC about this file by physical or electronic mail, you may
+// find current contact information at www.suse.com.
+
+use crate::http::BaseHTTPClient;
+
+use super::{
+    config::QuestionsConfig,
+    http_client::{HTTPClient as QuestionsHTTPClient, QuestionsHTTPClientError},
+};
+
+#[derive(Debug, thiserror::Error)]
+#[error("Error processing questions settings: {0}")]
+pub struct QuestionsStoreError(#[from] QuestionsHTTPClientError);
+
+type QuestionsStoreResult<T> = Result<T, QuestionsStoreError>;
+
+/// Loads and stores the questions settings from/to the HTTP API.
+pub struct QuestionsStore {
+    questions_client: QuestionsHTTPClient,
+}
+
+impl QuestionsStore {
+    pub fn new(client: BaseHTTPClient) -> Self {
+        Self {
+            questions_client: QuestionsHTTPClient::new(client),
+        }
+    }
+
+    pub async fn load(&self) -> QuestionsStoreResult<Option<QuestionsConfig>> {
+        Ok(None)
```

Comment:
this is a bit inconsistent with rest. With this the combination of `agama config load` and `agama config show` will omit answers, which I am not sure if it is intention or expected.

Reply:
It is intentional. I did not find a good way to export the list of questions. They are stored as trait objects if I am not wrong. I decided we can live with that for 16.0.  But I will reevaluate it if needed.

Reply:
I had a closer look to this issue. The main problem is that the elements in the [answer_strategies](https://github.com/agama-project/agama/blob/b191edc46d26a6446805e25fdd9c0641e4b47303/rust/agama-server/src/questions.rs#L159) should implement `Serialize` too, but this trait is not compatible with `dyn` (it is generic). We could rely on the [erased_serde crate](https://docs.rs/erased-serde/latest/erased_serde/), but I have not tried.

To be honest, given that we plan to do a major refactor of this part (dropping the D-Bus interface), I would consider a different implementation (e.g., using enums instead of trait objects).